agent/agent.go

```diff
@@ -3700,6 +3738,27 @@ func (a *Agent) ReloadConfig() error {
 	// breaking some existing behavior.
 	newCfg.NodeID = a.config.NodeID
 
+	//if auto reload is enabled, make sure we have the right certs file watched.
+	if autoReload {
+		for _, f := range []struct {
+			oldCfg string
+			newCfg string
+		}{{a.config.KeyFile, newCfg.KeyFile}, {a.config.CertFile, newCfg.CertFile}} {
+			if f.oldCfg != f.newCfg {
+				a.FileWatcher.Remove(f.oldCfg)
+				err = a.FileWatcher.Add(f.newCfg)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		if !reflect.DeepEqual(newCfg.StaticRuntimeConfig, a.config.StaticRuntimeConfig) {
+			a.logger.Warn("Static Runtime config has changed and need a manual config reload to be applied", "StaticRuntimeConfig", a.config.StaticRuntimeConfig, "StaticRuntimeConfig From file", newCfg.StaticRuntimeConfig)
```

Comment:
Is this clearer?

```suggestion
			a.logger.Warn("Static Runtime config has changed. Reloading config.", "StaticRuntimeConfig", a.config.StaticRuntimeConfig, "StaticRuntimeConfig From file", newCfg.StaticRuntimeConfig)
```

Reply:
Actually this is not accurate, when a `StaticRuntimeConfig` is updated we never reload it. This is by design to avoid deactivating important security features.